### DISCUSSION TITLE
Table field Number column needs same restrictions as Number field

### DISCUSSION THREAD

**Discussion by proimage** [craftcms/cms#9863] [Ideas]
tl;dr: See title.

Also, this started out as a bug report, before I reluctantly came to the dreaded realization that it wasn't actually a bug, it was a feature, and needed to be filed as an idea, because reasons. 🤷

Anyway, that's why it's formatted like this; deal with it. 😎 

### Description
For a multi-national client, I built a donation form widget. On the frontend, above the "Donation Amount" field there are a few "Suggested Amount" buttons:

![image](https://user-images.githubusercontent.com/1027523/134332858-7220d657-d89f-4636-bac1-148518bbe37d.png)

The values for these buttons are pre-defined in a backend table field, in a Number column. For most countries' branches, this worked fine—

Until the German branch.

![image](https://user-images.githubusercontent.com/1027523/134332624-b85ea2e5-afbb-496d-9ee3-691eb9410c7c.png)

Now, either they're suggesting very precise amounts for purposes such as buying some rather oddly-priced candy bars, or else they're trying to raise thousands of Euros to rescue slaves (hint: it's not for candy).

Now, I realize the Number field isn't a *currency* field per se, but it can certainly act as one, and the restrictions of how many decimal places are allowed are quite useful (including "none"). However, the Table field's Number column doesn't provide any of those sorts of restrictions.

Of course, the client should be trained to not put in thousands separators in number fields, yes. However, having the ability to show an error if a decimal point was used when it was not permitted would have given my client a pretty big hint that something was amiss... ;)

### Steps to reproduce

1, Be from a country that for whatever reason uses `.` as the thousands separator, and `,` as the decimal separator. 🤷‍♂️ 
2, Try to input a fully-formatted number into a number column of a Table field.

### Additional info

- Craft version: 3.7.3.2
- PHP version: 7.2.24
- Database driver & version: MySQL 8.0.25
- Plugins & versions:
  - Architect	2.4.3
  - Asset Rev	6.0.2
  - Bulk Edit	2.0.5
  - Button Box	2.0.4
  - Colorit	1.1.2.1
  - Colour Swatches	1.4.1.1
  - Cookies	1.1.15
  - DigitalOcean Spaces Volume	1.1.3
  - Entry Instructions	1.0.7
  - Feed Me	4.3.6
  - Field Manager	2.2.4
  - Freeform	3.11.9
  - Icon Picker	1.1.10
  - Image Resizer	2.1.0
  - Linkit	1.1.12.1
  - LJ Range Slider	1.0.0
  - Maps	3.9.1
  - Mix	1.5.2
  - Neo	2.11.1
  - oEmbed	1.3.9
  - Postmark	2.1.0
  - Redactor	2.8.7
  - Redactor Custom Styles	3.0.4
  - SEO	3.7.4
  - Simple Text	2.0.2
  - Super Table	2.6.8
  - Table Maker	2.0.1
  - Template Select	2.0.2
